modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/UpNextFragment.kt

```diff
@@ -175,20 +199,33 @@ class UpNextFragment : BaseFragment(), UpNextListener, UpNextTouchCallback.ItemT
         (recyclerView.itemAnimator as? SimpleItemAnimator)?.changeDuration = 0
 
         val multiSelectToolbar = view.findViewById<MultiSelectToolbar>(R.id.multiSelectToolbar)
-        multiSelectHelper.isMultiSelectingLive.observe(viewLifecycleOwner) {
-            multiSelectToolbar.isVisible = it
-            toolbar.isVisible = !it
+        multiSelectHelper.isMultiSelectingLive.observe(viewLifecycleOwner) { isMultiSelecting ->
+            val wasMultiSelecting = multiSelectToolbar.isVisible
+            multiSelectToolbar.isVisible = isMultiSelecting
+            toolbar.isVisible = !isMultiSelecting
+
+            /* Track only if not embedded. If it is an embedded fragment, then track only when in expanded state */
+            if (!isEmbedded || isEmbeddedExpanded()) {
+                if (isMultiSelecting) {
```

Comment:
`isEmbedded` is set to true for `UpNextFragment` created from the `PlayerContainerFragment` and shown when we swipe up/ down on the full-screen player screen. It gets created as soon as `PlayerContainerFragment` is created.

`multiSelectHelper` is a singleton and it seems there can be two instances of `UpNextFragment` that can be observing `isMultiSelectingLive` livedata. The other instance is created when we click the up-next icon on the mini-player or the full-screen player.

So that we don't track multi-select entered/ exited twice, I added the `if (!isEmbedded || isEmbeddedExpanded())`check.